dislib/regression/linear/base.py

```diff
@@ -105,88 +113,87 @@ def predict(self, x):
 
         for r_block in x._iterator(axis='rows'):
             blocks[0].append(
-                _predict(r_block._blocks, self.coef_, self.intercept_))
+                _predict(r_block._blocks, self._coef, self._intercept))
 
         return Array(blocks=blocks, top_left_shape=(x._top_left_shape[0], 1),
                      reg_shape=(x._reg_shape[0], 1), shape=(x.shape[0], 1),
                      sparse=x._sparse)
 
+    @property
+    def coef_(self):
+        validation.check_is_fitted(self, '_coef')
+        self._coef = compss_wait_on(self._coef)
+        return self._coef
 
-def _reduce(func, partials, arity):
-    while len(partials) > 1:
-        partials_chunk = partials[:arity]
-        partials = partials[arity:]
-        partials.append(func(*partials_chunk))
-    return partials[0]
+    @property
+    def intercept_(self):
+        validation.check_is_fitted(self, '_intercept')
+        self._intercept = compss_wait_on(self._intercept)
+        return self._intercept
```

Comment:
Wouldn't it be better to return coef and the intercept as a ds-array and let the user decide if they want to synchronize or not with collect?

Reply:
Done!